src/lazy/text/matched.rs

```diff
@@ -1050,9 +1139,48 @@ impl MatchedClob {
 
 #[cfg(test)]
 mod tests {
+    use std::str::FromStr;
+
+    use num_bigint::BigInt;
+
     use crate::lazy::bytes_ref::BytesRef;
     use crate::lazy::text::buffer::TextBufferView;
-    use crate::{Decimal, IonResult, Timestamp};
+    use crate::{Decimal, Int, IonResult, Timestamp};
+
+    #[test]
+    fn read_ints() -> IonResult<()> {
+        fn expect_int(data: &str, expected: impl Into<Int>) {
+            let expected: Int = expected.into();
+            let buffer = TextBufferView::new(data.as_bytes());
+            let (_remaining, matched) = buffer.match_int().unwrap();
+            let actual = matched.read(buffer).unwrap();
+            assert_eq!(
+                actual, expected,
+                "Actual didn't match expected for input '{}'.\n{:?}\n!=\n{:?}",
+                data, actual, expected
+            );
+        }
+
+        let tests = [
+            ("-5", Int::from(-5)),
```

Comment:
🗺️ Added int reading tests that could have caught the bug that discarded the sign information.